src/panels/lovelace/cards/hui-light-card.ts

```diff
@@ -0,0 +1,256 @@
+import { html, LitElement } from "@polymer/lit-element";
+import { styleMap } from "lit-html/directives/styleMap.js";
+import computeStateDomain from "../../../common/entity/compute_state_domain.js";
+import computeStateName from "../../../common/entity/compute_state_name.js";
+import stateIcon from "../../../common/entity/state_icon.js";
+import { jQuery } from "../../../resources/jquery";
+
+import "../../../components/ha-card.js";
+import "../../../components/ha-icon.js";
+import { roundSliderStyle } from "../../../resources/jquery.roundslider";
+
+import { HomeAssistant } from "../../../types.js";
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+
+const lightConfig = {
+  radius: 80,
+  step: 1,
+  circleShape: "pie",
+  startAngle: 315,
+  width: 5,
+  min: 0,
+  max: 100,
+  sliderType: "min-range",
+  lineCap: "round",
+  handleSize: "+10",
+  showTooltip: false,
+};
+
+interface Config extends LovelaceConfig {
+  entity: string;
+  name?: string;
+}
+
+export class HuiLightCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  static get properties() {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public getCardSize() {
+    return 2;
+  }
+
+  public setConfig(config: Config) {
+    if (!config.entity || config.entity.split(".")[0] !== "light") {
+      throw new Error("Specify an entity from within the light domain.");
+    }
+
+    this._config = config;
+  }
+
+  protected render() {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass!.states[this._config!.entity];
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card>
+        <div id="root">
+          <div id="light"></div>
+          <div id="tooltip">
+            <div class="icon-state">
+              <ha-icon
+                data-domain="${computeStateDomain(stateObj)}"
+                data-state="${stateObj.state}"
+                .icon="${stateIcon(stateObj)}"
+                style="${styleMap({
+                  filter: this._computeBrightness(stateObj),
+                  color: this._computeColor(stateObj),
+                })}"
+                @click="${this._handleClick}"
+              ></ha-icon>
+              <div class="brightness">${Math.round(
+                (stateObj.attributes.brightness / 254) * 100
+              ) || 0}%</div>
+              <div class="name">${this._config.name ||
+                computeStateName(stateObj)}</div>
+            </div>
+          </div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps) {
+    if (changedProps.get("hass")) {
+      return (
+        changedProps.get("hass").states[this._config!.entity] !==
+        this.hass!.states[this._config!.entity]
+      );
+    }
+    if (changedProps.get("_config")) {
```

Comment:
No need to check for this.

Reply:
If config was updated, we re-render.

Reply:
I thought you said to add this check in the other PR 🤔

Reply:
Not for `_config`, just for the state objects.